libs/JSystem/J3DGraphBase/J3DPacket.cpp

```diff
@@ -160,68 +107,55 @@ extern "C" extern u8 __GDCurrentDL[4];
 // Declarations:
 //
 
-/* 80312488-80312508 30CDC8 0080+00 1/1 1/1 0/0 .text newDisplayList__17J3DDisplayListObjFUl */
-#pragma push
-#pragma optimization_level 0
-#pragma optimizewithasm off
-asm void J3DDisplayListObj::newDisplayList(u32 param_0) {
-    nofralloc
-#include "asm/JSystem/J3DGraphBase/J3DPacket/newDisplayList__17J3DDisplayListObjFUl.s"
+J3DError J3DDisplayListObj::newDisplayList(u32 capacity) {
+    mCapacity = (capacity + 0x1F) & ~0x1F; // align
+    mpData[0] = new (0x20) char[mCapacity];
+    mpData[1] = new (0x20) char[mCapacity];
+    mSize = 0;
+    if (mpData[0] == NULL || mpData[1] == NULL)
+        return kJ3DError_Alloc;
+    return kJ3DError_Success;
 }
-#pragma pop
 
-/* 80312508-8031256C 30CE48 0064+00 1/1 1/1 0/0 .text newSingleDisplayList__17J3DDisplayListObjFUl
- */
-#pragma push
-#pragma optimization_level 0
-#pragma optimizewithasm off
-asm void J3DDisplayListObj::newSingleDisplayList(u32 param_0) {
-    nofralloc
-#include "asm/JSystem/J3DGraphBase/J3DPacket/newSingleDisplayList__17J3DDisplayListObjFUl.s"
+J3DError J3DDisplayListObj::newSingleDisplayList(u32 capacity) {
+    mCapacity = (capacity + 0x1F) & ~0x1F; // align
+    mpData[0] = new (0x20) char[mCapacity];
+    mpData[1] = mpData[0];
+    mSize = 0;
+    if (mpData[0] == NULL)
+        return kJ3DError_Alloc;
+    return kJ3DError_Success;
 }
-#pragma pop
 
 /* 8031256C-803125E4 30CEAC 0078+00 0/0 1/1 0/0 .text single_To_Double__17J3DDisplayListObjFv */
-#pragma push
-#pragma optimization_level 0
-#pragma optimizewithasm off
-asm void J3DDisplayListObj::single_To_Double() {
-    nofralloc
-#include "asm/JSystem/J3DGraphBase/J3DPacket/single_To_Double__17J3DDisplayListObjFv.s"
+J3DError J3DDisplayListObj::single_To_Double() {
+    if (mpData[0] == mpData[1]) {
+        mpData[1] = new (0x20) char[mCapacity];
+        if (mpData[1] == NULL)
+            return kJ3DError_Alloc;
+        memcpy(mpData[1], mpData[0], mCapacity);
+        DCStoreRange(mpData[1], mCapacity);
+    }
+
+    return kJ3DError_Success;
 }
-#pragma pop
 
-/* 803125E4-80312604 30CF24 0020+00 0/0 1/1 0/0 .text
- * setSingleDisplayList__17J3DDisplayListObjFPvUl               */
-#pragma push
-#pragma optimization_level 0
-#pragma optimizewithasm off
-asm void J3DDisplayListObj::setSingleDisplayList(void* param_0, u32 param_1) {
-    nofralloc
-#include "asm/JSystem/J3DGraphBase/J3DPacket/setSingleDisplayList__17J3DDisplayListObjFPvUl.s"
+void J3DDisplayListObj::setSingleDisplayList(void* pDLData, u32 size) {
+    mCapacity = (size + 0x1F) & ~0x1F;
```

Comment:
```suggestion
    mCapacity = ALIGN_NEXT(size, 0x20);
```
For that macro you need to import "global.h"